File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/ui/adapter/DetailsAdapter.java
package com.tilly.steven.stlbusarrivals.ui.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import com.tilly.steven.stlbusarrivals.Model.Details;
import com.tilly.steven.stlbusarrivals.R;
import java.util.ArrayList;
import java.util.Calendar;

/**
 * Created by Steven on 2016-01-28.
 */
public class DetailsAdapter extends ArrayAdapter<Details> {

    private LayoutInflater inflater;
    private String prediction;
    private Context ctx;

    public DetailsAdapter(Context context, int rowLayoutResourceId, ArrayList<Details> list) {
        super(context, rowLayoutResourceId, list);
        this.inflater = LayoutInflater.from(context);
        ctx = context;
    }

    @SuppressLint("DefaultLocale")
    @Override
    public View getView(int position, View view, ViewGroup parent) {

        ViewHolder holder;
        if (view != null) {
            holder = (ViewHolder) view.getTag();
        } else {
            view = inflater.inflate(R.layout.row_favorites, parent, false);
            holder = new ViewHolder();
            holder.routeName = (TextView) view.findViewById(R.id.tv_row_favorite_route);
            holder.stop = (TextView) view.findViewById(R.id.tv_row_favorite_stop);
            holder.arrivalTime = (TextView) view.findViewById(R.id.tv_row_favorite_arrival_time);
            holder.prediction = (TextView) view.findViewById(R.id.tv_row_favorite_prediction);
            view.setTag(holder);
        }

        Details item = getItem(position);
        prediction = item.getPrediction();

        if(prediction != null){
            Calendar c = Calendar.getInstance();
            int currentHour = c.get(Calendar.HOUR_OF_DAY);
            int currentMinutes = c.get(Calendar.MINUTE);
                int predictedMinutes = currentMinutes + Integer.valueOf(prediction);
                while(predictedMinutes >= 60){
                    currentHour++;
                    predictedMinutes = predictedMinutes - 60;
                }
                if(predictedMinutes < 10){
                    holder.arrivalTime.setText(currentHour + ":0"+ predictedMinutes);
                }else {
                    holder.arrivalTime.setText(currentHour + ":"+ predictedMinutes);
                }
            holder.prediction.setText(ctx.getString(R.string.next_bus, prediction));

        }else{
            holder.arrivalTime.setText(ctx.getString(R.string.no_bus));
        }
        holder.routeName.setText(item.getRouteName());
        holder.stop.setText(item.getStopName());
        return view;
    }

    static class ViewHolder {
        TextView routeName;
        TextView stop;
        TextView arrivalTime;
        TextView prediction;
    }
}

File: mobile/src/main/java/com/tilly/steven/stlbusarrivals/ui/adapter/DetailsAdapter.kt
package com.tilly.steven.stlbusarrivals.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.tilly.steven.stlbusarrivals.R
import com.tilly.steven.stlbusarrivals.model.Details
import java.util.*

/**
 * Created by Steven on 2016-01-28.
 */
class DetailsAdapter(private val ctx: Context, rowLayoutResourceId: Int, list: List<Details>) : ArrayAdapter<Details>(ctx, rowLayoutResourceId, list) {

    private val inflater: LayoutInflater = LayoutInflater.from(ctx)

    @SuppressLint("DefaultLocale")
    override fun getView(position: Int, view1: View?, parent: ViewGroup): View {
        var view = view1

        val holder: ViewHolder
        if (view != null) {
            holder = view.tag as ViewHolder
        } else {
            view = inflater.inflate(R.layout.row_favorites, parent, false)!!
            holder = ViewHolder()
            holder.routeName = view.findViewById<View>(R.id.tv_row_favorite_route) as TextView
            holder.stop = view.findViewById<View>(R.id.tv_row_favorite_stop) as TextView
            holder.arrivalTime = view.findViewById<View>(R.id.tv_row_favorite_arrival_time) as TextView
            holder.prediction = view.findViewById<View>(R.id.tv_row_favorite_prediction) as TextView
            view.tag = holder
        }

        val item = getItem(position)
        val prediction = item?.prediction

        if (prediction != null) {
            val c = Calendar.getInstance()
            var currentHour = c.get(Calendar.HOUR_OF_DAY)
            val currentMinutes = c.get(Calendar.MINUTE)
            var predictedMinutes = currentMinutes + Integer.valueOf(prediction)
            while (predictedMinutes >= 60) {
                currentHour++
                predictedMinutes -= 60
            }
            if (predictedMinutes < 10) {
                holder.arrivalTime.text = "$currentHour:0$predictedMinutes"
            } else {
                holder.arrivalTime.text = "$currentHour:$predictedMinutes"
            }
            holder.prediction.text = ctx.getString(R.string.next_bus, prediction)

        } else {
            holder.arrivalTime.text = ctx.getString(R.string.no_bus)
        }
        holder.routeName.text = item?.routeName
        holder.stop.text = item?.stopName
        return view
    }

    internal class ViewHolder {
        lateinit var routeName: TextView
        lateinit var stop: TextView
        lateinit var arrivalTime: TextView
        lateinit var prediction: TextView
    }
}
